packages/caliper-core/test/worker/caliper-worker.js

```diff
@@ -161,5 +167,302 @@ describe('Caliper worker', () => {
             await worker.executeRound(mockTestMessage);
             validateCallsAndWarnings(4);
         });
+
+        [5, 10].forEach(numberOfTxs => {
+            it(`should run ${numberOfTxs} transactions and wait for completion when no errors occur`, async () => {
+                const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+                await worker.prepareTest(mockTestMessage);
+
+                mockTestMessage.getNumberOfTxs.returns(numberOfTxs);
+                mockTestMessage.getRoundDuration.returns(null);
+
+                mockWorkload.submitTransaction.resetHistory();
+                mockStats.getTotalSubmittedTx.resetHistory();
+                mockStats.getTotalFinishedTx.resetHistory();
+                mockStats.getCumulativeTxStatistics.resetHistory();
+
+                let submittedTx = 0;
+                let finishedTx = 0;
+
+                // Stub the methods
+                mockStats.getTotalSubmittedTx.callsFake(() => submittedTx);
+                mockStats.getTotalFinishedTx.callsFake(() => finishedTx);
+                mockStats.getCumulativeTxStatistics.returns({});
+
+                worker.internalTxObserver.getCurrentStatistics = () => mockStats;
+
+                mockWorkload.submitTransaction.callsFake(async () => {
+                    submittedTx += 1;
+                    finishedTx += 1;
+                    return Promise.resolve();
+                });
+
+                await worker.executeRound(mockTestMessage);
+
+                sinon.assert.callCount(mockWorkload.submitTransaction, numberOfTxs);
+                sinon.assert.calledOnce(deactivateMethod);
+                sinon.assert.calledOnce(mockRate.end);
+                sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+                sinon.assert.called(mockConnector.releaseContext);
+            });
+        });
+
+        [5, 10].forEach(numberOfTxs => {
+            it(`should handle errors while running ${numberOfTxs} transactions`, async () => {
```

Comment:
should handle errors doesn't tell me as a reader what should happen when the error occurs. Please be explicit about what happens when an error occurs. In this case the round is stopped.